### PR TITLE
Add some missing bounds checks.

### PR DESCRIPTION
**Pull request by goffrie**
Ran a fuzzer and found a few places where we were panicking instead of returning errors.

### CODE REVIEW

File: src/frame/headers.rs [hunk 1]
```diff
@@ -153,6 +153,9 @@ impl Headers {
 
         // Read the padding length
         if flags.is_padded() {
+            if src.len() < 1 {
+                return Err(Error::MalformedMessage);
+            }
             // TODO: Ensure payload is sized correctly
```
Comment (seanmonstar) [inline src/frame/headers.rs]: Maybe there's more involved, but wanted to check: does this addition essentially complete this TODO?

File: src/proto/streams/streams.rs [hunk 2]
```diff
@@ -322,6 +322,10 @@ where
         let last_stream_id = frame.last_stream_id();
         let err = frame.reason().into();
 
+        if actions.recv.max_stream_id() < last_stream_id {
```
Comment (seanmonstar) [inline src/proto/streams/streams.rs]: I got a bit confused with this, until I went digging into `recv` to read the comments about `max_stream_id`. Whatcha think if there was a comment right here just saying to the effect of "if a new GOAWAY has a higher stream id than a previous GOAWAY, that's bad"?

File: src/frame/headers.rs (hunk 1 above)
Comment (goffrie) [reply src/frame/headers.rs]: Seems like it; I removed the comment.

File: src/proto/streams/streams.rs (hunk 2 above)
Comment (goffrie) [reply src/proto/streams/streams.rs]: :+1:

**Comment by carllerche** [comment]
Thanks @goffrie!

Looks like the fuzzing has been submitted in another PR. I'm good with this!

**Pull request merged**